### ISSUE TITLE
Any plan to seperate mimalloc modifications?

### ISSUE THREAD

**Issue by cyyever** [colesbury/nogil#34]
I have noticed that mimalloc integrated in nogil has some custom code. Can we refactor the custom modifications into other files such that users can try difference versions of mimalloc? This is convenience in case there are bug fixes in further mimalloc versions.

**Comment by colesbury** [comment]
No, there are changes to mimalloc data structures and existing functions so they can't be separated into files. The changes are necessary for the garbage collection support and for the thread-safe access to dicts and lists that mostly avoids lock acquisitions.

**Comment by thedrow** [comment]
Could we upstream the patches somehow?

**Comment by colesbury** [comment]
The changes are specific to nogil Python, so I don't think it would make sense to adopt them in upstream mimalloc.

**Comment by zhangkaizhao** [comment]
The core team of Python is trying to [bpo-46657: Add mimalloc memory allocator](https://bugs.python.org/issue46657) to Python 3.11 and it seems that it gets a little improvement on performance.
> mimalloc sources will vendored without any option to use system libraries. Python's mimalloc requires several non-standard compile-time flags. In the future Python may extend or modify mimalloc for heap walking and nogil, too.
